app/controllers/queue_items_controller.rb

```diff
@@ -0,0 +1,32 @@
+class QueueItemsController < ApplicationController 
+  before_action :require_login
+
+  def index
+    @queue_items = current_user.queue_items
+  end
+  
+  def create
+    @video = Video.find(params[:video_id])
+    @queue_item = @video.queue_items.new 
+    @queue_item.user = User.find(params[:user_id])
+    @queue_item.position = queue_item_count
```

Comment:
Also instead of 4 lines 9-12 you should check solution and make 1 line from this.